crates/sui-node/src/lib.rs

```diff
@@ -1594,7 +1591,14 @@ impl SuiNode {
         }))
     }
 
-    async fn reexecute_pending_consensus_certs(
+    // TODO: Fix this function.
+    // We only attempt to re-execute pending consensus transactions that is:
+    // 1. Fast-path transaction
+    // 2. Has signed the effects
```

Comment:
I guess the original idea of this function is to not change the response when querying the effects of a transaction, once the effects has been returned to a full node? Arguably after the full node gathers a quorum of effects sigs, it is ok for individual validators to temporarily not return the effects, as long as what the validator returns is not treated as permanent failures and the transactions gets included in checkpoints eventually.

Reply:
Fullnode usually call validators through the handle_certificate interface, which will eventually execute it again and return the same response. The only exception is that if a fullnode calls the `transaction_info` interface, which I believe is extremely rare (I remember we only did it when trying to resolve equivocation of sorts)